### ISSUE TITLE
reqwest not found on armv7l-unknown-linux-gnueabihf and aarch64-unknown-linux-musl

### ISSUE THREAD

**Issue by classabbyamp** [nvarner/typst-lsp#234]
- **Component**: 
    - Build
- **LSP version**: 0.9.0
- **OS version and name**: Void Linux

- I am on the [latest](https://github.com/nvarner/typst-lsp/tags) stable version of the extension/LSP.
- I have searched the [issues](https://github.com/nvarner/typst-lsp/issues) of this repo and believe that this is not a duplicate.

## Issue

The reqwest feature division in 1e368ed9759671c5f4658dc37b186a1f9bdc9ff8 is incomplete, as it is missing some platforms. Simply adding more features as people report them missing is unsustainable, however. I'd suggest letting people choose their own TLS backend, either by passing the features directly (`--features reqwest/rustls-tls`) or passing it down for them:
```toml
[features]
rustls = ["reqwest/rustls-tls"]
```

### Logs

```rs
   Compiling typst-lsp v0.9.0 (/builddir/typst-lsp-0.9.0)
error[E0432]: unresolved import `reqwest`
 --> src/workspace/package/external/repo.rs:7:5
  |
7 | use reqwest::{Client, Url};
  |     ^^^^^^^ use of undeclared crate or module `reqwest`

warning: unused import: `futures::TryStreamExt`
 --> src/workspace/package/external/repo.rs:6:5
  |
6 | use futures::TryStreamExt;
  |     ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0433]: failed to resolve: use of undeclared crate or module `reqwest`
  --> src/workspace/package/external/repo.rs:94:24
   |
94 |     NotFound(#[source] reqwest::Error),
   |                        ^^^^^^^ use of undeclared crate or module `reqwest`

error[E0433]: failed to resolve: use of undeclared crate or module `reqwest`
  --> src/workspace/package/external/repo.rs:96:13
   |
96 |     Network(reqwest::Error),
   |             ^^^^^^^ use of undeclared crate or module `reqwest`

error[E0433]: failed to resolve: use of undeclared crate or module `reqwest`
   --> src/workspace/package/external/repo.rs:105:39
    |
105 |         let err = match Self::io_as::<reqwest::Error>(err) {
    |                                       ^^^^^^^ use of undeclared crate or module `reqwest`

error[E0433]: failed to resolve: use of undeclared crate or module `reqwest`
   --> src/workspace/package/external/repo.rs:137:53
    |
137 |         if err.get_ref().is_some_and(|err| err.is::<reqwest::Error>()) {
    |                                                     ^^^^^^^ use of undeclared crate or module `reqwest`

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
warning: `typst-lsp` (bin "typst-lsp") generated 1 warning
error: could not compile `typst-lsp` (bin "typst-lsp") due to 5 previous errors; 1 warning emitted
```

**Comment by nvarner** [comment]
This should be fixed in the v0.9.1 release being built now. Per your recommendation, the TLS backend can be configured via features. The default is `rustls`. Thanks for the report and advice on fixing it!

**Comment by classabbyamp** [comment]
thanks!
